### PR TITLE
Feat/124 donation modal

### PR DESCRIPTION
**Pull request by SiboYang**
# Summary
Implemented a modal that will appear after after clicking the Donate in the dropdown menu(full screen in mobile), 
TODO: add links for the button, add the exit button, I will finish it when I get up.

## Test Plan

Changed the screen size
Screen Shot 2021-03-11 at 12 20 12 AM (2)
Screen Shot 2021-03-11 at 12 20 37 AM

## Related Issues
#124 

Which issues does this PR resolve/work on?

### CODE REVIEW

File: frontend/src/DonationModal/DonationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      marginTop: theme.spacing(5),
+      display:'block',
+      
+  }
+}));
+
+function DonationModal(props: IDonationModalProps) {
+  const classes = useStyles();
+  const theme = useTheme();
+  const fullScreen = useMediaQuery(theme.breakpoints.down("sm"));
```
Comment (tedspare) [inline frontend/src/DonationModal/DonationModal.tsx:42]: nice!!! so glad you figured this out - it will help others

**Review by tedspare — APPROVED** [review]
I know it's not finished yet but your code looks really good to me so far! Keep it up 😄

File: frontend/src/DonationModal/DonationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              </div>
+              
+            </div>
+          }
+        </DialogContentText>
+      </DialogContent>
+      <Button
+        className={classes.button}
+        size="large"
+        variant="contained"
```
Comment (tedspare) [inline frontend/src/DonationModal/DonationModal.tsx:94]: add `disableElevation`

**Review by BlancheFrancheterre — APPROVED** [review]
Nothing to add to Ted's comments. Looks great !!

File: frontend/src/App/App.tsx [hunk 3]
```diff
@@ -68,6 +70,10 @@ function App() {
 
   const openSignin = () => setSigningIn(true);
 
+  const handleDonationOpen = () => setDonateOpen(true);
```
[the remark this answers is not in the document]
Comment (SiboYang) [reply frontend/src/App/App.tsx]: Are the ones used in signinForm need to change as well?

[the remark this answers is not in the document]
Comment (tedspare) [reply frontend/src/App/App.tsx]: eventually but it's okay for now!
our codebase isn't totally standardized but, whenever someone comes up with a nice solution (like the media query below), I tell everyone about it going forward

**Review by tedspare — APPROVED** [review]
LGTM! good to merge

**Comment by tedspare** [comment]
could move colour and accent on Sign In to Donate, to encourage users to donate
maybe add a 💸 or 🙏 too?

Screen Shot 2021-03-11 at 9 11 36 AM

**Comment by tedspare** [comment]
could add some margin below the Donate button

Screen Shot 2021-03-11 at 9 15 07 AM

**Comment by SiboYang** [comment]
> could move colour and accent on Sign In to Donate, to encourage users to donate
> maybe add a 💸 or 🙏 too?
> 
> Screen Shot 2021-03-11 at 9 11 36 AM

That's a good idea! Just to make sure, by accent do you mean make the text bold?

**Pull request merged**